maro/cli/inspector/cim_dashboard.py

```diff
@@ -0,0 +1,473 @@
+
+    x_axis_single = list(range(0, matrix_len))
+    x_axis = [x_axis_single] * matrix_len
+    y_axis = [[row[col] for row in x_axis] for col in range(len(x_axis[0]))]
+    # Convert this grid to columnar data expected by Altair
+    source = pd.DataFrame({
+        "Dest_Port": np.array(x_axis).ravel(),
+        "Start_Port": np.array(y_axis).ravel(),
+        "Count": np.array(b).ravel()})
+    hot_map = alt.Chart(source).mark_rect().encode(
+        x="Dest_Port:O",
+        y="Start_Port:O",
+        color="Count:Q",
+        tooltip=["Dest_Port", "Start_Port", "Count"]
+    ).properties(
+        width=700,
+        height=380
+    )
+    st.altair_chart(hot_map)
+
+
+def _generate_top_k_summary(data: pd.DataFrame, snapshot_index: int, name_conversion: pd.DataFrame):
+    """Generate CIM top 5 summary.
+
+    Args:
+        data (dataframe): Data of current snapshot.
+        snapshot_index (int): Selected snapshot index.
+        name_conversion (dataframe): Relationship between index and name.
+    """
+    data_acc = data[data["frame_index"] == snapshot_index].reset_index(drop=True)
+    data_acc["fulfillment_ratio"] = list(
+        map(lambda x, y: float("{:.4f}".format(x / (y + 1 / 1000))), data_acc["acc_fulfillment"],
+            data_acc["acc_booking"]))
+
+    data_acc["port name"] = list(map(lambda x: name_conversion.loc[int(x[6:])][0], data_acc["name"]))
+    helper.render_h3_title("Select Top k")
+    top_number = st.select_slider("", list(range(0, 10)))
+    top_attributes = CIMItemOption.acc_info + ["fulfillment_ratio"]
+    for item in top_attributes:
+        helper.generate_by_snapshot_top_summary("port name", data_acc, int(top_number), item, snapshot_index)
+
+
+def _generate_down_pooling_sample(down_pooling_num: int, start_epoch: int, end_epoch: int) -> list:
+    """Generate down pooling list based on original data and down pooling rate.
+        This function aims to generate epoch samples.
+        No requirements for the sampled data.
+
+    Args:
+        down_pooling_num(int): Calculated length of sample data list.
+        start_epoch(int): Start index of sample data.
+        end_epoch(int): End index of sample data.
+
+    Returns:
+        list: sample data list.
+    """
+    down_pooling_len = math.floor(1 / down_pooling_num)
+    down_pooling_range = list(range(start_epoch, end_epoch, down_pooling_len))
+    if end_epoch not in down_pooling_range:
+        down_pooling_range.append(end_epoch)
+
+    return down_pooling_range
+
+
+def _get_sampled_epoch_range(epoch_num: int, sample_ratio: float) -> list:
+    """For inter plot, generate sampled data list based on range & sample ratio
+
+    Args:
+        epoch_num(int): Number of snapshot folders.
+        sample_ratio(float): Sampling ratio.
+        e.g. If sample_ratio = 0.3, and sample data range = [0,10],
+        down_pooling_list = [0, 0.3, 0.6, 0.9]
+        down_pooling_range = [0, 3, 6, 9]
+
+    Returns:
+        list: list of sampled data index
```

Comment:
use ``List[int]`` instead, more specific, better.

Also,
Start with a capital letter, end with a period. Same to other doc-strings.